cypress/pageobjects/image.po.ts

```diff
@@ -1,45 +1,183 @@
-import PagePo from '@/utils/components/page.po';
+import { Constants } from "@/constants/constants";
 import LabeledInputPo from '@/utils/components/labeled-input.po';
+import { VolumePage } from "@/pageobjects/volume.po";
+
+const constants = new Constants();
+const volumes = new VolumePage();
 
 interface ValueInterface {
   namespace?: string,
-  name?: string,
+  name: string,
   description?: string,
-  url: string,
+  url?: string,
+  size?: string,
+  path?: string,
+  labels?: any,
 }
 
-export default class ImagePage extends PagePo {
-  static url: string = `${Cypress.env('baseUrl')}dashboard/c/local/harvester/harvesterhci.io.virtualmachineimage`
+export class ImagePage {
+  name() {
+    return new LabeledInputPo('.namespace-select > .labeled-input', `:contains("Name")`)
+  }
+
+  url() {
+    return new LabeledInputPo('.labeled-input', `:contains("URL")`);
+  }
 
-  constructor() {
-    super(ImagePage.url);
+  description() {
+    return new LabeledInputPo('.labeled-input', `:contains("Description")`)
   }
 
   public goToList() {
-    cy.visit(ImagePage.url);
+    // cy.get('.nav').contains('Images').click();
+    cy.visit(constants.imagePage);
     cy.intercept('GET', '/v1/harvester/harvesterhci.io.virtualmachineimages').as('goToImageList');
-    cy.wait('@goToImageList')
-    expect(cy.url().should('eq', ImagePage.url));
+    cy.wait('@goToImageList');
+    cy.url().should('eq', constants.imagePage);
   }
 
   public goToCreate() {
-    this.goToList()
-    cy.contains('Create').click()
+    this.goToList();
+    cy.contains('Create').click();
+  }
+
+  public goToEdit(name: string) {
+    this.goToList();
+    cy.get('.search').type(name);
```

Comment:
The selector should be stored as a class member not selected just based on a CSS class
```
private search = '.search';
cy.get(this.search).type(name);
```

Reply:
done